src/world/entity.rs

```diff
@@ -383,20 +406,38 @@ impl Generation {
     /// Panics in debug mode if it's not alive.
     fn die(&mut self) {
         debug_assert!(self.is_alive());
-        self.0 = -self.0;
+        self.0 = NonZeroU32::new(-self.id() as u32).map(Generation);
```

Comment:
Can't this underflow and therefore panic in debug builds?

Reply:
Sure can, so yeah that should be corrected.

Reply:
Which operation can overflow?

Reply:
The assert should ensure that `id()` is positive and the only overflow case is where `id()` is i32::MIN so I think this one is correct. AFAIK signed-ness casts are always no-ops in rust so that should be OK as well

Reply:
Ok; I can accept the `assert!` making the negation safe, but I'd like to see where casts from signed to unsigned are supposed to be defined as bitcasts.

Reply:
Never mind; found it: https://doc.rust-lang.org/reference/expressions/operator-expr.html#semantics

Reply:
`id()` can only ever be a positive number by virtue of the fact that it is built from a `NonZeroU32`.  Won't this always panic?

Reply:
Hm ok but it can only ever be positive or 0.  I think we still have a problem.